### ISSUE TITLE
[oracle] unparsable section on max(...) KEEP (DENSE_RANK FIRST/LAST ... ORDER BY ...)

### ISSUE THREAD

**Issue by GHRF** [sqlfluff/sqlfluff#7947]
### Search before asking

- [x] I searched the [issues](https://github.com/sqlfluff/sqlfluff/issues) and found no similar issues.

### What Happened

While deploying sqlfluff on a project with hundreds of Oracle queries, I am running into a few unparsable sections even though the queries : 
- run, and yield expected results 
- are syntactically correct according to the Oracle documentation
- do not seem to use any discouraged/deprecated syntax

It seems like it's because of the syntax for `KEEP (DENSE_RANK FIRST/LAST ... ORDER BY ...)`, which is quite unique !

### Expected Behaviour

When using the `KEEP (DENSE_RANK FIRST/LAST ... ORDER BY ...)` syntax, sqlfluff should not raise a PRS error and should be able to parse this clause, as is defined in the Oracle documentation (under FIRST and LAST):
- https://docs.oracle.com/en/database/oracle/oracle-database/21/sqlrf/FIRST.html
- https://docs.oracle.com/en/database/oracle/oracle-database/21/sqlrf/LAST.html

### Observed Behaviour

Since the queries in my project can be lenghty, I went back to the documentation to create minimal examples to reproduce.

When linting the example queries from the oracle documentation, sqlfluff raises PRS errors too, which confirms my suspicion. Looking at the sqlfluff source code, I do not see the DENSE_RANK keyword being used to match any segments.
 
I am providing everything needed in the next sections to reproduce :)

### How to reproduce

Here are some minimal queries to reproduce, as well as the parsed result in the human output format: 

**Queries:** [keep_dense_rank.sql](https://github.com/user-attachments/files/28890625/keep_dense_rank.sql), based on the [oracle documentation](https://docs.oracle.com/en/database/oracle/oracle-database/21/sqlrf/FIRST.html)
```
SELECT
    employee_id,
    max(salary) KEEP (DENSE_RANK FIRST ORDER BY hire_date) AS first_salary
FROM employees
GROUP BY employee_id;

SELECT
    employee_id,
    max(salary) KEEP (DENSE_RANK LAST ORDER BY hire_date) AS last_salary
FROM employees
GROUP BY employee_id;

SELECT
    employee_id,
    max(salary) KEEP (DENSE_RANK FIRST ORDER BY hire_date NULLS LAST) AS first_known_salary
FROM employees
GROUP BY employee_id;
```

**Result:** [keep_dense_rank_PRS.txt](https://github.com/user-attachments/files/28890629/keep_dense_rank_PRS.txt) - obtained witht the following command:

```
sqlfluff parse keep_dense_rank.sql --format human --write-output keep_dense_rank_PRS.txt
```

### Dialect

oracle

### Version

sqlfluff, version 4.2.2

### Configuration

This minimal configuration is enough to reproduce.

```
[sqlfluff]
dialect = oracle
```

### Are you willing to work on and submit a PR to address the issue?

- [x] Yes I am willing to submit a PR!

### Code of Conduct

- [x] I agree to follow this project's [Code of Conduct](https://github.com/sqlfluff/sqlfluff/blob/main/CODE_OF_CONDUCT.md)
